cmd/terramate/cli/cloud.go

```diff
@@ -287,138 +302,165 @@ func (c *cli) detectCloudMetadata() {
 		Token:      ghToken,
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), defaultGithubTimeout)
-	defer cancel()
+	if ghCommit, err := getGithubCommit(&ghClient, ghRepo, headCommit); err == nil {
+		setGithubCommitMetadata(md, ghCommit)
+	} else {
+		logger.Warn().
+			Err(err).
+			Msg("failed to retrieve commit information from GitHub API")
+	}
 
-	headCommit := c.prj.headCommit()
-	pulls, err := ghClient.PullsForCommit(ctx, ghRepo, headCommit)
-	if err != nil {
+	if pulls, err := getGithubPR(&ghClient, ghRepo, headCommit); err == nil {
+		if len(pulls) == 0 {
+			logger.Warn().
+				Msg("no pull request associated with HEAD commit")
+
+			return
```

Comment:
returning here will not sync *GithubActions* metadata when no pull request is associated with the commit.

Reply:
Good catch, I looked through the previous version and I had actually not fully understood the intention behind the specific return points before. I changed it so the behaviour should be the same again.